lte/gateway/c/session_manager/SessionState.cpp

```diff
@@ -460,7 +500,19 @@ bool SessionState::apply_update_criteria(SessionStateUpdateCriteria& uc) {
       return false;
     }
   }
-
+  // Converged 5G rules to install
+  for (const auto& rule : uc.PdrList_to_install) {
+    int32_t Pdr_index;
+    Pdr_index = pdr_index(rule.pdr_id());
+    if (Pdr_index != -1) {
+      // Update the existing value
+      PdrList_.at(Pdr_index) = rule;
+    } else {
+      // Insert the rule
+      PdrList_.push_back(rule);
+    }
+  }
```

Comment:
same here, using a map, this would be one liner

Reply:
While integrating MVC code with policydb, we can change the above vector to hashmaps.